modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/services/v2/checker/excel/ErrorAnalysisService.kt

```diff
@@ -48,20 +52,23 @@ class PropagatedErrorsService(
         val references = graph.successors(cell)
         for (reference in references) {
             if (!visited.contains(reference)) {
-                findPropagatedErrors(reference, errors, perrors, visited)
+                findErrors(reference)
             }
         }
 
         // Graph Deconstruction
         // eval cell again and compare again with solution cell
         evaluator.evaluateInCell(workbookCell)
         if (!cellEqualsSolution(cell, workbookCell)) {
-            handleService?.getHandlers(When.ONERROR)?.forEach { handler -> handler.handle(ErrorAnalysisContext()) }
             errors.add(cell) // add to original errors set
+            handleService?.getHandlers(When.ONERROR)
+                ?.forEach { handler -> handler.handle(ErrorAnalysisContext(errors, perrors, cell)) }
```

Comment:
Instead of looping here, I would call a function on the `HandlerService' that runs all handlers of that type.